templates/deployment.yaml

```diff
@@ -25,42 +29,100 @@ spec:
         - name: remote-network-agent
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
-          readinessProbe:
-            httpGet:
-              path: /health/readiness
-              port: 8080
+          {{- if .Values.resources }}
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
+          {{- end }}
           livenessProbe:
             httpGet:
               path: /health/liveness
               port: 8080
           startupProbe:
-            initialDelaySeconds: 60
-            periodSeconds: 10
+            periodSeconds: 1
             timeoutSeconds: 1
             successThreshold: 1
-            failureThreshold: 6
+            failureThreshold: 60
             httpGet:
               path: /health/liveness
               port: 8080
-          {{$podData := dict "Values" .Values "Type" "pod"}}
-          {{- include "armory-remote-network-agent.request-limits" $podData | nindent 8 | trim }}
+          {{- if .Values.config }}
+          volumeMounts:
+            - name: overrides-config
+              mountPath: /opt/go-application/config/rna-overrides.yaml
+              subPath: rna-overrides
+          {{- end }}
           env:
+            - name: APPLICATION_NAME
+              value: "rna"
+            - name: APPLICATION_ENVIRONMENT
+              value: {{ .Values.environment | default "prod" }}
+            - name: APPLICATION_VERSION
+              value: "{{ .Values.image.tag | default .Chart.AppVersion }}"
+            - name: LOGGER_TYPE
+              value: "{{ .Values.log.type }}"
+            - name: LOGGER_LEVEL
+              value: "{{ .Values.log.level }}"
+            - name: DISABLE_COLORS
+              value: "{{ .Values.log.disableColor }}"
+            {{- if not (empty .Values.sandboxClusterId) }}
+            - name: agent_sandboxClusterId
+              value: {{ .Values.sandboxClusterId }}
+            {{- end }}
+            - name: oidc_clientId
+              value: "{{ include "armory-remote-network-agent.client-id" . }}"
+            - name: oidc_clientSecret
```

Comment:
out of curiosity - shouldnt client id and client secret be loaded from secrets? i mean - simple kubectl get pod would reveal all the environment values, but if that was configured with envFromSecret, it wouldn't be so easily visible. on the other hand - if one can run kubectl get po, there is nothing preventing him from doing kubectl exec and inspecting the actual env ...

Reply:
Yes you are correct, in the readme and values file we highly encourage the use of encrypted tokens

![image](https://user-images.githubusercontent.com/711726/201171586-991d402f-ca8e-4453-ac14-39757664d2f4.png)

https://github.com/armory-io/remote-network-agent-helm-chart/blob/e9cf9962d86f578d20e69e9ff1e607ea564cdb56/values.yaml#L1-L28

So the value that will be injected here will be 

```
encrypted:secrets-manager!r:us-west-2!s:some-aws-sm-secret-name!k:client-secret
```

or

```
encrypted:secrets-manager!r:us-west-2!s:some-aws-sm-secret-name!k:client-secret
```

:point_up: these are converted to the actual values at run time